backend/metadata_writer/src/metadata_writer.py

```diff
@@ -393,6 +397,9 @@ def is_kfp_v2_pod(pod) -> bool:
 
                 pods_with_written_metadata[obj.metadata.name] = None
 
-        except Exception as e:
-            import traceback
-            print(traceback.format_exc())
```

Comment:
do we still want to catch other exceptions that are not client timeouts?

Reply:
Originally I wanted to keep this, but removed after a comment by @thesuperzapper, because but with the catch-all except clause removed it will be easier for a user to notice that something is wrong, because Kubernetes will restart metadata-writer pod and its restart counter will be incremented. 

In a case that an exception is more severe then a timeout and really blocks metadata-writer from doing its job, it is probaly better to let user now by crashing and restarting it. With the catch-all except clause in the case it will just infinetely repeat the loop so metadata-writer will appear as normally working but will not do anything.

See the thread after https://github.com/kubeflow/pipelines/pull/11361#discussion_r1831813126